samples/api/hello_triangle_1_3/hello_triangle_1_3.cpp

```diff
@@ -0,0 +1,1290 @@
+		vkGetPhysicalDeviceProperties(physical_device, &device_properties);
+
+		uint32_t major_version = VK_API_VERSION_MAJOR(device_properties.apiVersion);
+		uint32_t minor_version = VK_API_VERSION_MINOR(device_properties.apiVersion);
+
+		if ((major_version < 1) || (major_version == 1 && minor_version < 3))
+		{
+			LOGW("Physical device '{}' does not support Vulkan 1.3, skipping.", device_properties.deviceName);
+			continue;
+		}
+
+		// Find a queue family that supports graphics and presentation
+		uint32_t queue_family_count = 0;
+		vkGetPhysicalDeviceQueueFamilyProperties(physical_device, &queue_family_count, nullptr);
+
+		std::vector<VkQueueFamilyProperties> queue_family_properties(queue_family_count);
+		vkGetPhysicalDeviceQueueFamilyProperties(physical_device, &queue_family_count, queue_family_properties.data());
+
+		for (uint32_t i = 0; i < queue_family_count; i++)
+		{
+			VkBool32 supports_present = VK_FALSE;
+			vkGetPhysicalDeviceSurfaceSupportKHR(physical_device, i, context.surface, &supports_present);
+
+			if ((queue_family_properties[i].queueFlags & VK_QUEUE_GRAPHICS_BIT) && supports_present)
+			{
+				context.graphics_queue_index = i;
+				break;
+			}
+		}
+
+		if (context.graphics_queue_index >= 0)
+		{
+			context.gpu = physical_device;
+			break;
+		}
+	}
+
+	if (context.graphics_queue_index < 0)
+	{
+		throw std::runtime_error("Failed to find a suitable GPU with Vulkan 1.3 support.");
+	}
+
+	uint32_t device_extension_count;
+
+	VK_CHECK(vkEnumerateDeviceExtensionProperties(context.gpu, nullptr, &device_extension_count, nullptr));
+
+	std::vector<VkExtensionProperties> device_extensions(device_extension_count);
+
+	VK_CHECK(vkEnumerateDeviceExtensionProperties(context.gpu, nullptr, &device_extension_count, device_extensions.data()));
+
+	// Since this sample has visual output, the device needs to support the swapchain extension
+	std::vector<const char *> required_device_extensions{VK_KHR_SWAPCHAIN_EXTENSION_NAME};
+
+	if (!validate_extensions(required_device_extensions, device_extensions))
+	{
+		throw std::runtime_error("Required device extensions are missing");
+	}
+
+	// Query for Vulkan 1.3 features
+	VkPhysicalDeviceFeatures2                       device_features2{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2};
+	VkPhysicalDeviceVulkan13Features                vulkan13_features{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_3_FEATURES};
+	VkPhysicalDeviceExtendedDynamicStateFeaturesEXT extended_dynamic_state_features{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_FEATURES_EXT};
+	device_features2.pNext  = &vulkan13_features;
+	vulkan13_features.pNext = &extended_dynamic_state_features;
+
+	vkGetPhysicalDeviceFeatures2(context.gpu, &device_features2);
+
+	// Enable required Vulkan 1.3 features
+	vulkan13_features.dynamicRendering                   = VK_TRUE;
+	vulkan13_features.synchronization2                   = VK_TRUE;
+	extended_dynamic_state_features.extendedDynamicState = VK_TRUE;
```

Comment:
Not sure if that's what we want. It's a 1.3 sample and those features are a requirement. So imo it's fine to set them that way, as device creation will then fail with a proper error message.

Reply:
I'm sure, that's what we want.
I think, the samples here should show how to do it right.
You're right, the spec says that you **can** check for extension presence and you **can** check for feature support. You don't have to, as `vkCreateDevice` would fail with some appropriate error code if you ask for something not supported.
But here, you're asking for each and every feature (listed by your structure chain) that's supported by that device (plus some that might actually not be supported). You should not do that, as it might result in sub-optimal performance or 
unnecessary memory consumption.
Instead, you should just ask for the extensions you need and the features you need.
And I think, it would be a good habit to check for their support before just trying to create the device.

Reply:
I agree that i should only enable the features that i am using. I will update the code